win32/src/win32evtlog.i

```diff
@@ -824,26 +933,120 @@ static PyObject *PyEvtSeek(PyObject *self, PyObject *args, PyObject *kwargs)
 }
 PyCFunction pfnPyEvtSeek = (PyCFunction) PyEvtSeek;
 
-// @pyswig str|EvtRender|Formats an event into XML text
+// @pyswig <o PyEVT_HANDLE>|EvtCreateRenderContext|Creates a render context
 // @comm Accepts keyword args
-// @comm Rendering event values (Flags=EvtRenderEventValues) is not currently supported
+static PyObject* PyEvtCreateRenderContext(PyObject *self, PyObject *args, PyObject *kwargs)
+{
+	static char *keywords[] = {"Flags", NULL};
+	DWORD flags = EvtRenderContextSystem;
+
+	if (!PyArg_ParseTupleAndKeywords(args, kwargs, "k:EvtCreateRenderContext", keywords,
+		&flags))	// @pyparm int|Flags||EvtRenderContextSystem or EvtRenderContextUser. EvtRenderContextValues not currently supported
+		return NULL;
+
+	EVT_HANDLE ret = NULL;
+	Py_BEGIN_ALLOW_THREADS
+	ret = EvtCreateRenderContext(0, NULL, flags);
+	Py_END_ALLOW_THREADS
+
+	if (ret == NULL) {
+		return PyWin_SetAPIError("EvtCreateRenderContext", GetLastError());
+	}
+
+	return PyWinObject_FromEVT_HANDLE(ret);
+}
+PyCFunction pfnPyEvtCreateRenderContext = (PyCFunction) PyEvtCreateRenderContext;
+
+// Returns a list of Event Values in the same order as returned by the EvtRender function
+static PyObject *RenderEventValues(EVT_HANDLE render_context, EVT_HANDLE event)
+{
+	PyObject* ret = NULL;
+	if (render_context == NULL) {
+		PyWin_SetAPIError("EvtRender - Invalid render context for EvtRenderEventValues");
+		return NULL;
+	}
+
+	DWORD allocated_size = 0;
+	DWORD returned_size = 0;
+	DWORD prop_count = 0;
+	PEVT_VARIANT variants = NULL;
+	BOOL bsuccess = 0;
+	DWORD err = ERROR_SUCCESS;
+
+	Py_BEGIN_ALLOW_THREADS
+	bsuccess = EvtRender(render_context, event, EvtRenderEventValues, allocated_size, variants, &returned_size, &prop_count);
+	Py_END_ALLOW_THREADS
+
+	// bsuccess should always be false here, because we call it initially to get
+	// the size of the buffer
+	if (!bsuccess) {
+		err = GetLastError();
+		if (err != ERROR_INSUFFICIENT_BUFFER) {
+			PyWin_SetAPIError("EvtRender", err);
+			goto cleanup;
+		}
+
+		// allocate buffer size
+		allocated_size = returned_size;
+		variants = (PEVT_VARIANT)malloc(allocated_size);
+		if (variants == NULL) {
+			PyErr_NoMemory();
+			goto cleanup;
+		}
+
+		Py_BEGIN_ALLOW_THREADS
+		bsuccess = EvtRender(render_context, event, EvtRenderEventValues, allocated_size, variants, &returned_size, &prop_count);
+		Py_END_ALLOW_THREADS
+	}
+
+	if (!bsuccess) {
+		err = GetLastError();
```

Comment:
it's very minor, but there's no need to explicitly call GetLastError - just don't pass a 2nd param.